src/quafu/users/userapi.py

```diff
@@ -1,85 +1,98 @@
-from ..utils.platform import get_homedir
+import json
 import os
+
 import requests
-import json
-from urllib import parse
+
+from ._quafu_url import QUAFU_URL
 from .exceptions import UserError
+from ..utils.platform import get_homedir
 
 
 class User(object):
-    def __init__(self):
-        self.apitoken = ""
-        self._url = "http://quafu.baqis.ac.cn/"
-        
-        if self.check_account_data():
-            self.load_account()
-            
+    def __init__(self, api_token: str = None, token_dir: str = None):
+        """
+        Initialize user account and load backend information.
+
+        :param api_token: if provided
+        :param token_dir: where api token is found or saved
+        """
+        self.url = QUAFU_URL
+        self._available_backends = {}
+
+        if token_dir is None:
+            self.token_dir = get_homedir() + "/.quafu/"
+        else:
+            self.token_dir = token_dir
+
+        if api_token is None:
+            self.api_token = self._load_account_token()
+        else:
+            self.api_token = api_token
+
         self.backends_api = "qbackend/get_backends/"
         self.chip_api = "qbackend/scq_get_chip_info/"
         self.exec_api = "qbackend/scq_kit/"
         self.exec_async_api = "qbackend/scq_kit_asyc/"
         self.exec_recall_api = "qbackend/scq_task_recall/"
         self.priority = 2
 
-    def check_account_data(self):
-        homedir = get_homedir()
-        file_dir = homedir + "/.quafu/"
-        if not os.path.exists(file_dir):
-            print("Your user information is not configured. Remember to configure it by User.save_apitoken(<your api_token>)")
-            return False
-        return True
-        # TODO: Check if the user's basic data file is formated
-
-    def save_apitoken(self, apitoken):
+    def save_apitoken(self, apitoken=None):
         """
-        Save your apitoken associate your Quafu account.
+        Save api-token associate your Quafu account.
         """
-        self.apitoken = apitoken
-        homedir = get_homedir()
-        file_dir = homedir + "/.quafu/"
+        if apitoken is not None:
+            import warnings
+            warnings.warn("The argument 'apitoken' in this function will be deprecated "
+                          "in the future, please set api token by providing 'api_token' "
+                          "or 'token_dir' when initialize User()."
+                          )
+            self.api_token = apitoken
+
+        file_dir = self.token_dir
         if not os.path.exists(file_dir):
             os.mkdir(file_dir)
         with open(file_dir + "api", "w") as f:
-            f.write(self.apitoken+"\n")
-            f.write("http://quafu.baqis.ac.cn/")
+            f.write(self.api_token + "\n")
+            f.write(QUAFU_URL)
 
-    def load_account(self):

```

Comment:
This api is already exposed to users before, I thinks we should maintain it for now. Be careful when modifying public methods.